tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -2797,5 +2980,563 @@ llvm::Optional<Value> convertTFConv2DCommon(
       .getResult();
 }
 
+// Lowers Gather operators to a sequence of TOSA ops.
+llvm::Optional<Value> convertGatherOp(PatternRewriter& rewriter, Operation* op,
+                                      Value result_value, Value params_value,
+                                      Value indices_value, int32_t batch_dims,
+                                      int32_t axis)
+
+{
+  auto result_type = result_value.getType().dyn_cast<RankedTensorType>();
+  auto params_type = params_value.getType().dyn_cast<RankedTensorType>();
+  auto indices_type = indices_value.getType().dyn_cast<RankedTensorType>();
+
+  if (!result_type) return llvm::None;
```

Comment:
nit: one if statement with multiple || clauses for compactness.

Reply:
Thanks. Will update in next round.